client/mainmenu/CMainMenu.cpp

```diff
@@ -729,25 +732,51 @@ CLoadingScreen::CLoadingScreen(ImagePath background)
 	CCS->musich->stopMusic(5000);
 
 	const auto& conf = CMainMenuConfig::get().getConfig()["loading"];
-	const auto& nameConfig = conf["name"];
 
-	AnimationPath animationPath;
-	if (nameConfig.isVector() && !nameConfig.Vector().empty())
-		animationPath = AnimationPath::fromJson(*RandomGeneratorUtil::nextItem(nameConfig.Vector(), CRandomGenerator::getDefault()));
+	// Load background
+	const auto& backgroundConfig = conf["background"];
+	if (backgroundConfig.isVector() && !backgroundConfig.Vector().empty())
+		this->background = std::make_shared<CPicture>(ImagePath::fromJson(*RandomGeneratorUtil::nextItem(backgroundConfig.Vector(), CRandomGenerator::getDefault())));
```

Comment:
No need to explicitly write `this` unless necessary
```suggestion
		background = std::make_shared<CPicture>(ImagePath::fromJson(*RandomGeneratorUtil::nextItem(backgroundConfig.Vector(), CRandomGenerator::getDefault())));
```

Reply:
Can't be compiled without **this->**